Mathlib/Algebra/Order/GroupWithZero/Bounds.lean

```diff
@@ -18,14 +18,12 @@ open Set
 lemma BddAbove.range_comp_of_nonneg {α β γ : Type*} [Nonempty α] [Preorder β] [Zero β] [Preorder γ]
     {f : α → β} {g : β → γ} (hf : BddAbove (range f)) (hf0 : 0 ≤ f)
     (hg : MonotoneOn g {x : β | 0 ≤ x}) : BddAbove (range (fun x => g (f x))) := by
-  have hg' : BddAbove (g '' (range f)) := by
-    apply hg.map_bddAbove (by rintro x ⟨a, rfl⟩; exact hf0 a)
-    · obtain ⟨b, hb⟩ := hf
-      use b, hb
-      simp only [mem_upperBounds, mem_range, forall_exists_index, forall_apply_eq_imp_iff] at hb
-      exact le_trans (hf0 Classical.ofNonempty) (hb Classical.ofNonempty)
-  change BddAbove (range (g ∘ f))
-  simpa only [Set.range_comp] using hg'
+  rw [← Function.comp_def, Set.range_comp]
+  apply hg.map_bddAbove (by rintro x ⟨a, rfl⟩; exact hf0 a)
+  obtain ⟨b, hb⟩ := hf
+  use b, hb
+  simp only [mem_upperBounds, mem_range, forall_exists_index, forall_apply_eq_imp_iff] at hb
+  exact le_trans (hf0 Classical.ofNonempty) (hb Classical.ofNonempty)
```

Comment:
Hmm, I personally slightly prefer the older proof here for readability's sake, since it makes it clear what the proof idea is. But the `change` still feels suboptimal: do you think there's a third option which avoids the `change` but still keeps the `have` block?

Reply:
The `suffices` is all good!